contracts/staking/contracts/src/StakingProxy.sol

```diff
@@ -95,6 +97,44 @@ contract StakingProxy is
         emit ReadOnlyModeSet(readOnlyMode);
     }
 
+    /// @dev Batch executes a series of calls to the exchange contract.
+    /// @param data An array of data that encodes a sequence of functions to
+    ///             call in the staking contracts.
+    function batchExecute(bytes[] memory data)
+        public
```

Comment:
Can this be made `external`, or do we get an internal compiler error?

Reply:
I made this public so that other proxy functions could make use of it (like a `stakeAndMove` function). It seems like we'll be leaving those functions out though, so I'll try to change this.